content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,221 +11,196 @@ weight: 10
+to learn more about CSI. 
 
-For more details, see the [`azureFile` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_file/README.md).
+[FlexVolume](#flexvolume) is an another mechanism for storage plugins. Unlike CSI, FlexVolume is
+more closely tied to Kubernetes and its storage implementation. FlexVolume was developed before CSI.
 
-#### azureFile CSI migration
+Storage plugin interfaces:
 
-{{< feature-state for_k8s_version="v1.15" state="alpha" >}}
+   * [CSI](#csi)
+   * [FlexVolume](#flexvolume)
+   * [persistentVolumeClaim](#persistentvolumeclaim)
 
-The `CSIMigration` feature for `azureFile`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `file.csi.azure.com` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Azure File CSI
-Driver](https://github.com/kubernetes-sigs/azurefile-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAzureFile`
-alpha features must be enabled.
 
-### cephfs
+### External Volume Types
 
-A `cephfs` volume allows an existing CephFS volume to be
-mounted into your Pod. Unlike `emptyDir`, which is erased when a pod is
-removed, the contents of a `cephfs` volume are preserved and the volume is merely
-unmounted. This means that a `cephfs` volume can be pre-populated with data, and
-that data can be shared between pods. The `cephfs` volume can be mounted by multiple
-writers simultaneously.
+The Kubernetes binary includes some volume types for communicating with externally managed volumes, such as AWS EBS or Azure Disk Storage. These volume types are described as in-tree storage plugins. 
 
-{{< note >}}
-You must have your own Ceph server running with the share exported before you can use it.
-{{< /note >}}
+The CSI was introduced to standardize connectivity to externally managed volumes, such as cloud platform storage. However, Kubernetes still includes many volume types for connecting to specific cloud storage providers. 
 
-See the [CephFS example](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/volumes/cephfs/) for more details.
+External Volume Types:
+   * [awsElasticBlockStore](#awselasticblockstore)
+   * [azureDisk](#azuredisk)
+   * [azureFile](#azurefile)
+   * [cinder](#cinder)
+   * [flocker](#flocker)
+   * [gcePersistentDisk](#gcepersistentdisk)
+   * [portworxVolume](#portworxvolume)
+   * [quobyte](#quobyte)
+   * [storageos](#storageos)
+   * [scaleIO](#scaleio)
+   * [vsphereVolume](#vspherevolume)
 
-### cinder
+## Specify subPaths on volumes {#using-subpath}
 
-{{< note >}}
-Kubernetes must be configured with the OpenStack cloud provider.
-{{< /note >}}
+Sometimes, it is useful to share one volume for multiple uses in a single pod. The `volumeMounts.subPath` property specifies a sub-path inside the referenced volume instead of its root.
```

Comment:
`volumeMounts.subPath` can be useful even if the volume is mounted just once in a Pod. For example, if the volume layout is determined externally from Kubernetes.

In this example we mount the same volume twice, and that's useful to show that it's possible.

Reply:
I reworded this to indicate the multiple mounts in a pod is an example.